mm2src/mm2_bitcoin/serialization/src/reader.rs

```diff
@@ -54,18 +54,27 @@ pub enum CoinVariant {
     LBC,
     Standard,
     Qtum,
+    // Todo: https://github.com/KomodoPlatform/atomicDEX-API/issues/1345
+    BTC,
 }
 
 impl CoinVariant {
     pub fn is_qtum(&self) -> bool { matches!(self, CoinVariant::Qtum) }
 
     pub fn is_lbc(&self) -> bool { matches!(self, CoinVariant::LBC) }
+
+    pub fn is_btc(&self) -> bool { matches!(self, CoinVariant::BTC) }
 }
 
 pub fn coin_variant_by_ticker(ticker: &str) -> CoinVariant {
-    match ticker {
-        "LBC" => CoinVariant::LBC,
-        _ => CoinVariant::Standard,
+    if ticker == "BTC" || ticker.contains("BTC-") || ticker.contains("BTC_") {
+        // "BTC", "BTC-segwit", "tBTC", "tBTC-segwit", etc..
+        CoinVariant::BTC
+    } else if ticker == "LBC" || ticker.contains("LBC-") || ticker.contains("LBC_") {
+        // "LBC", "LBC-segwit", etc..
+        CoinVariant::LBC
+    } else {
+        CoinVariant::Standard
```

Comment:
what about replacing `coin_variant_by_ticker` with this? to avoid having boring `if/else` statements

```rust
fn coin_variant_by_ticker(ticker: &str) -> CoinVariant {
    // "BTC", "BTC-segwit", "tBTC", "tBTC-segwit", etc..
    let btc_ticker = ticker == "BTC" || ticker.contains("BTC-") || ticker.contains("BTC_");
    // "LBC", "LBC-segwit", etc..
    let lbc_ticker = ticker == "LBC" || ticker.contains("LBC-") || ticker.contains("LBC_");

    match (btc_ticker, lbc_ticker) {
        (btc @ true, _) => CoinVariant::BTC,
        (_, lbc @ true) => CoinVariant::LBC,
        _ => CoinVariant::Standard,
    }
}
```

until at least when this issue is resolved done #1345

Reply:
I think the below is even a better way of doing this :)
```
pub fn coin_variant_by_ticker(ticker: &str) -> CoinVariant {
    match ticker {
        // "BTC", "BTC-segwit", "tBTC", "tBTC-segwit", etc..
        t if t == "BTC" || t.contains("BTC-") || t.contains("BTC_") => CoinVariant::BTC,
        // "LBC", "LBC-segwit", etc..
        t if t == "LBC" || t.contains("LBC-") || t.contains("LBC_") => CoinVariant::LBC,
        _ => CoinVariant::Standard,
    }
}
```
I will also remove `coin_variant_by_ticker` in favor of `impl From<&str> for CoinVariant`